test/test.js

```diff
@@ -579,13 +588,13 @@ suite('include-fragment-element', function () {
       div.hidden = false
     }, 0)
 
-    return load
-      .then(() => when(div.firstChild, 'include-fragment-replaced'))
-      .then(() => {
-        assert.equal(loadCount, 1, 'Load occured too many times')
-        assert.equal(document.querySelector('include-fragment'), null)
-        assert.equal(document.querySelector('#replaced').textContent, 'hello')
-      })
+    const replacedPromise = when(div.firstChild, 'include-fragment-replaced')
+
+    return load.then(replacedPromise).then(() => {
```

Comment:
The original code was assuming that `load` would return before `include-fragment-replaced` fired. We do have some careful code to match the microtasking order of an `<img>` load, but I believe in this case the test was making a faulty assumption by awaiting the entire `load` before registering the the handler using `when()`. Let me know if I need to fix the test or code in another way.

Reply:
Looks like the test is wrong, I think your alteration is correct.